app/views/split_checkout/_details.html.haml

```diff
@@ -77,7 +77,7 @@
     - ship_method_description = nil
 
     - selected_shipping_method ||= @shipping_methods[0].id if @shipping_methods.length == 1
-    - @shipping_methods.each do |shipping_method|
+    - @allowed_shipping_methods.each do |shipping_method|
```

Comment:
The `selected_shipping_method` can still point to an invalid method. It doesn't really matter here because it won't be selected in this input but it's confusing. It's also the only reference to `@shipping_methods`. So we don't need that any more.